### ISSUE TITLE
Fatal Exception: java.lang.NullPointerException (com.pichillilorenzo.flutter_inappwebview.InAppWebView.InputAwareWebView.onFocusChanged)

### ISSUE THREAD

**Issue by fif1973** [pichillilorenzo/flutter_inappwebview#379]
## Environment

**Flutter version:** 1.12.13+hotfix+8 => 1.17.3
**Plugin version:**  3.0.0 => 3.3.0+3
**Android version:**  5.1 => 10
**iOS version:**
**Xcode version:**  
**Device information:**  many (Samsung, Hisense, Asus, Huawei,...)

## Description

Report from CrashAnalytics

## Steps to reproduce

I don't know ? Stacktraces from Firebase/CrashAnalytics

## Stacktrace/Logcat 

Some examples:

Marque : HUAWEI
Modèle : Mate 20 lite
Version : 8.1.0
```
Fatal Exception: java.lang.NullPointerException: Attempt to invoke virtual method 'void io.flutter.embedding.engine.systemchannels.PlatformViewsChannel.invokeViewFocused(int)' on a null object reference
       at io.flutter.plugin.platform.PlatformViewsController$1.lambda$createPlatformView$0$PlatformViewsController$1(PlatformViewsController.java:123)
       at io.flutter.plugin.platform.-$$Lambda$PlatformViewsController$1$yCLpCrkCk5PKTxnfIMt18jB15HI.onFocusChange(:4)
       at android.view.View.onFocusChanged(View.java:6997)
       at android.webkit.WebView.onFocusChanged(WebView.java:2872)
       at com.pichillilorenzo.flutter_inappwebview.InAppWebView.InputAwareWebView.onFocusChanged(InputAwareWebView.java:232)
       at android.view.View.handleFocusGainInternal(View.java:6669)
       at android.view.ViewGroup.handleFocusGainInternal(ViewGroup.java:733)
       at android.view.View.requestFocusNoSearch(View.java:10935)
       at android.view.View.requestFocus(View.java:10914)
       at android.view.ViewGroup.requestFocus(ViewGroup.java:3179)
       at android.webkit.WebView.-wrap12()
       at android.webkit.WebView$PrivateAccess.super_requestFocus(WebView.java:2398)
       at com.android.webview.chromium.WebViewChromium.requestFocus(PG:8)
       at android.webkit.WebView.requestFocus(WebView.java:2900)
       at android.view.View.requestFocus(View.java:10881)
       at android.view.View.requestFocus(View.java:10823)
       at com.pichillilorenzo.flutter_inappwebview.InAppWebView.InAppWebViewClient.onPageFinished(InAppWebViewClient.java:223)
       at Dn.b(PG:2)
       at iu0.handleMessage(PG:66)
       at android.os.Handler.dispatchMessage(Handler.java:109)
       at android.os.Looper.loop(Looper.java:166)
       at android.app.ActivityThread.main(ActivityThread.java:7555)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:469)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:963)
```

Marque : samsung
Modèle : Galaxy Tab A (2016)
Version : 8.1.0
```
Fatal Exception: java.lang.NullPointerException: Attempt to invoke virtual method 'void io.flutter.embedding.engine.systemchannels.PlatformViewsChannel.invokeViewFocused(int)' on a null object reference
       at io.flutter.plugin.platform.PlatformViewsController$1.lambda$createPlatformView$0$PlatformViewsController$1(PlatformViewsController.java:123)
       at io.flutter.plugin.platform.-$$Lambda$PlatformViewsController$1$yCLpCrkCk5PKTxnfIMt18jB15HI.onFocusChange(:4)
       at android.view.View.onFocusChanged(View.java:7614)
       at android.webkit.WebView.onFocusChanged(WebView.java:2872)
       at com.pichillilorenzo.flutter_inappwebview.InAppWebView.InputAwareWebView.onFocusChanged(InputAwareWebView.java:232)
       at android.view.View.handleFocusGainInternal(View.java:7286)
       at android.view.ViewGroup.handleFocusGainInternal(ViewGroup.java:733)
       at android.view.View.requestFocusNoSearch(View.java:11593)
       at android.view.View.requestFocus(View.java:11572)
       at android.view.ViewGroup.requestFocus(ViewGroup.java:3374)
       at android.webkit.WebView.-wrap12()
       at android.webkit.WebView$PrivateAccess.super_requestFocus(WebView.java:2398)
       at com.android.webview.chromium.WebViewChromium.requestFocus(chromium-Monochrome.aab-stable-410408370:8)
       at android.webkit.WebView.requestFocus(WebView.java:2900)
       at android.view.View.requestFocus(View.java:11539)
       at android.view.View.requestFocus(View.java:11481)
       at com.pichillilorenzo.flutter_inappwebview.InAppWebView.InAppWebViewClient.onPageFinished(InAppWebViewClient.java:223)
       at bn.b(chromium-Monochrome.aab-stable-410408370:2)
       at Ox0.handleMessage(chromium-Monochrome.aab-stable-410408370:66)
       at android.os.Handler.dispatchMessage(Handler.java:106)
       at android.os.Looper.loop(Looper.java:164)
       at android.app.ActivityThread.main(ActivityThread.java:7000)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:441)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1408)
```

Marque : Xiaomi
Modèle : Redmi Note 8
Version : 9
```
Fatal Exception: java.lang.NullPointerException: Attempt to invoke virtual method 'void io.flutter.embedding.engine.systemchannels.PlatformViewsChannel.invokeViewFocused(int)' on a null object reference
       at io.flutter.plugin.platform.PlatformViewsController$1.lambda$createPlatformView$0$PlatformViewsController$1(PlatformViewsController.java:123)
       at io.flutter.plugin.platform.-$$Lambda$PlatformViewsController$1$yCLpCrkCk5PKTxnfIMt18jB15HI.onFocusChange(:4)
       at android.view.View.onFocusChanged(View.java:7276)
       at android.webkit.WebView.onFocusChanged(WebView.java:2781)
       at com.pichillilorenzo.flutter_inappwebview.InAppWebView.InputAwareWebView.onFocusChanged(InputAwareWebView.java:232)
       at android.view.View.handleFocusGainInternal(View.java:6945)
       at android.view.ViewGroup.handleFocusGainInternal(ViewGroup.java:795)
       at android.view.View.requestFocusNoSearch(View.java:11559)
       at android.view.View.requestFocus(View.java:11533)
       at android.view.ViewGroup.requestFocus(ViewGroup.java:3232)
       at android.webkit.WebView.access$1001(WebView.java:101)
       at android.webkit.WebView$PrivateAccess.super_requestFocus(WebView.java:2298)
       at com.android.webview.chromium.WebViewChromium.requestFocus(chromium-Monochrome.aab-stable-410408373:8)
       at android.webkit.WebView.requestFocus(WebView.java:2809)
       at android.view.View.requestFocus(View.java:11500)
       at android.view.View.requestFocus(View.java:11442)
       at com.pichillilorenzo.flutter_inappwebview.InAppWebView.InAppWebViewClient.onPageFinished(InAppWebViewClient.java:223)
       at bn.b(chromium-Monochrome.aab-stable-410408373:2)
       at Ox0.handleMessage(chromium-Monochrome.aab-stable-410408373:66)
       at android.os.Handler.dispatchMessage(Handler.java:106)
       at android.os.Looper.loop(Looper.java:201)
       at android.app.ActivityThread.main(ActivityThread.java:6864)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:547)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:873)
```

Marque : samsung
Modèle : Galaxy A5
Version : 6.0.1
```
Fatal Exception: java.lang.NullPointerException: Attempt to invoke virtual method 'void io.flutter.embedding.engine.systemchannels.PlatformViewsChannel.invokeViewFocused(int)' on a null object reference
       at io.flutter.plugin.platform.PlatformViewsController$1.lambda$createPlatformView$0$PlatformViewsController$1(PlatformViewsController.java:123)
       at io.flutter.plugin.platform.-$$Lambda$PlatformViewsController$1$yCLpCrkCk5PKTxnfIMt18jB15HI.onFocusChange(lambda)
       at android.view.View.onFocusChanged(View.java:6273)
       at android.webkit.WebView.onFocusChanged(WebView.java:2552)
       at com.pichillilorenzo.flutter_inappwebview.InAppWebView.InputAwareWebView.onFocusChanged(InputAwareWebView.java:232)
       at android.view.View.handleFocusGainInternal(View.java:5999)
       at android.view.ViewGroup.handleFocusGainInternal(ViewGroup.java:732)
       at android.view.View.requestFocusNoSearch(View.java:9072)
       at android.view.View.requestFocus(View.java:9051)
       at android.view.ViewGroup.requestFocus(ViewGroup.java:3053)
       at android.webkit.WebView.access$1001(WebView.java:300)
       at android.webkit.WebView$PrivateAccess.super_requestFocus(WebView.java:2163)
       at com.android.webview.chromium.WebViewChromium.requestFocus(chromium-SystemWebViewGoogle.apk-stable-410408350:8)
       at android.webkit.WebView.requestFocus(WebView.java:2580)
       at android.view.View.requestFocus(View.java:9018)
       at android.view.View.requestFocus(View.java:8997)
       at com.pichillilorenzo.flutter_inappwebview.InAppWebView.InAppWebViewClient.onPageFinished(InAppWebViewClient.java:223)
       at Sb.b(chromium-SystemWebViewGoogle.apk-stable-410408350:2)
       at Ut.handleMessage(chromium-SystemWebViewGoogle.apk-stable-410408350:65)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:148)
       at android.app.ActivityThread.main(ActivityThread.java:7331)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1230)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1120)
```

Last week, this append 107 times for 90 users of my app

**Comment by VerachadW** [comment]
Any updates on this issue? We also have similar errors on in our production app.

**Comment by pichillilorenzo** [comment]
Yes, it should be fixed starting from version `3.4.0`. The latest release is `3.4.0+2`.
